superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx

```diff
@@ -171,19 +171,26 @@ class DatasourceControl extends React.PureComponent {
 
   onDatasourceSave = datasource => {
     this.props.actions.changeDatasource(datasource);
-    const timeCol = this.props.form_data?.granularity_sqla;
```

Comment:
Hi @geido , there is a [util](https://github.com/apache/superset/blob/master/superset-frontend/packages/superset-ui-chart-controls/src/utils/getTemporalColumns.ts) for this use case.